File: src/main/java/com/warrantybee/api/services/implementations/OtpService.java
package com.warrantybee.api.services.implementations;

import com.warrantybee.api.models.User;
import com.warrantybee.api.services.interfaces.IOtpService;
import org.springframework.stereotype.Service;
import java.security.SecureRandom;

/**
 * Implementation of {@link IOtpService} that generates
 * a cryptographically secure, 6-digit one-time password (OTP).
 */
@Service
public class OtpService implements IOtpService {

    private final SecureRandom _secureRandom;

    /**
     * Constructs a new OtpService and initializes a {@link SecureRandom} instance.
     */
    public OtpService() {
        _secureRandom = new SecureRandom();
    }

    /**
     * Generates a new, cryptographically secure, 6-digit OTP.
     * The number is padded with leading zeros to ensure it is always 6 digits.
     *
     * @return A {@code String} representing the generated 6-digit OTP.
     */
    @Override
    public String generate() {
        int otp = _secureRandom.nextInt(1_000_000);
        return String.format("%06d", otp);
    }
}

File: src/WarrantyBee.Application/Services/OtpService.cs
using System.Security.Cryptography;
using WarrantyBee.Shared.Infrastructure.Abstractions;
using WarrantyBee.Application.Abstractions.Services;

namespace WarrantyBee.Application.Services;

/// <summary>
/// Service for generating One-Time Passwords (OTPs).
/// </summary>
public class OtpService : IOtpService
{
    /// <summary>
    /// Generates a random 6-digit OTP.
    /// </summary>
    /// <returns>A string representing the generated OTP.</returns>
    public string Generate()
    {
        var otp = RandomNumberGenerator.GetInt32(0, 1_000_000);
        return otp.ToString("D6");
    }
}
